cli/lib/command/internal_command_rebase.dart

```diff
@@ -72,9 +73,28 @@ class InternalCommandRebase extends InternalCommand {
 
     final rebaseOnto = targetNode.line.branchNameOrCommitHash();
 
+    final branchesForRebase = current.localBranchNamesInOrderForRebase();
+
+    if (context.repositorySettings != null) {
+      final branches =
+          branchesForRebase
+              .map((node) => {'node': node.node, 'parent': node.parent ?? ''})
+              .toList();
+
+      final rebaseData = RebaseData(
+        hasTheirsFlag: hasTheirsFlag,
+        hasOursFlag: hasOursFlag,
+        rebaseOnto: rebaseOnto,
+        branches: branches,
+      );
+
+      context.repositorySettings!.rebaseData.value = rebaseData;
```

Comment:
Let's not use repository settings to save rebase data, let's create new file `rebase.json` in `.git/info/stax`. We will use absence or presence of the file as flag if we are currently in rebase.